app/classes/api/observation_api.rb

```diff
@@ -6,27 +6,23 @@ class API
   class ObservationAPI < ModelAPI
     self.model = Observation
 
-    self.high_detail_page_length = 10
-    self.low_detail_page_length  = 100
+    self.high_detail_page_length = 100
+    self.low_detail_page_length  = 1000
     self.put_page_length         = 1000
     self.delete_page_length      = 1000
 
     self.high_detail_includes = [
-      :comments,
-      { images: :license },
-      :location,
-      :name,
-      { namings: [:name, { votes: :user }] },
       :collection_numbers,
+      { comments: :user },
+      :external_links,
       { herbarium_records: :herbarium },
-      :sequences,
-      :user
-    ]
-
-    self.low_detail_includes = [
```

Comment:
I'm surprised this is going away essentially for the same reasons as mentioned above.

Reply:
Low detail queries were never supposed to include related objects.  The stated rule of thumb was that low-detail queries returned only what was in that table.  If you want more (e.g., the owner's name instead of just the owner's user_id), then you should be asking for the high-detail response.  I was including the name and location before because I couldn't imagine any use case where a low-detail response without name and location would be useful.  But now, thanks to the location name and consensus name (text_name) being cached in the observations table, we can still include those in the low-detail responses.  What that means is you can get pretty much everything you need for simple uses from low-detail: who, when, where, what and even the thumbnail image.

It is worth noting that as far as the DB is concerned, both the low-detail and high-detail queries are the same order.  The way the includes (eager loading) works is that it loads the main table, e.g., observations first, then in a second query it loads all the owners, then in a third it loads all the consensus names, then namings, then votes, then images, then comments, etc. etc. each table one at a time.  So it's all still O(N).  The big difference in run time is from the render, not the DB.  And now the render time is drastically reduced, so it's okay to request high-detail responses as much as you want.

So, yes, there are some minor changes that could potentially break existing API users' code, but the functionality is all still there.  It should be very easy to adapt to the new (more consistent and cleaner) format.

Reply:
I guess it's ok with me.  The API stuff is really your thing (as is probably obvious from my comments) so it's up to you.  I know I've had to deal with grumpy users whenever I've just made changes to an API pretty much no matter how small without versioning the API.  APIs are very different from human readable UIs.  The problem is that if anything breaks in their code due to some assumption they've made, they probably don't have the time to fix it at the moment so this often means the just stop using the API rather than finding time to fix it.  This is one of the reasons that old versions of APIs linger on often for many years.  In some ways the value of an API is related to how stable it is.  That being said if there aren't many users and you feel confident that either their code won't break with these change or they'll have the time and inclination to get things fixed if they do, then I'm fine with moving forward.